### PR TITLE
[flutter_tools] dont allow creating package name that is invalid

### LINKED ISSUE #52363: Remove noop androidx flag from flutter create
Once https://github.com/flutter/flutter/pull/52340 is merged, there will be a noop `androidx` flag, which is meant to be removed after the next stable release.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

According to https://dart.dev/tools/pub/pubspec#name , uppercase is not permitted for package names.

Also fix https://github.com/flutter/flutter/issues/52363 cause why not

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 1]
```diff
@@ -132,12 +132,9 @@ class CreateCommand extends FlutterCommand {
       defaultsTo: 'kotlin',
       allowed: <String>['java', 'kotlin'],
     );
-    // TODO(egarciad): Remove this flag. https://github.com/flutter/flutter/issues/52363
     argParser.addFlag(
-      'androidx',
```
Comment (cyanglaz) [inline packages/flutter_tools/lib/src/commands/create.dart:137]: cc @blasten To make sure it is ready to be removed.

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 2]
```diff
@@ -132,12 +132,9 @@ class CreateCommand extends FlutterCommand {
...
+      'skip-checks',
```
Comment (cyanglaz) [inline packages/flutter_tools/lib/src/commands/create.dart:136]: Why did we need this flag? Is there a scenario where we want to not validate project names?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/create.dart:136]: Just for our own integration tests

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/create.dart:136]: Technically we can't stop people from changing the name after the fact, and our tooling needs to work with arbitrary names

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 1 above)
Comment (blasten) [reply packages/flutter_tools/lib/src/commands/create.dart:137]: it should be fine at this point

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 2 above)
Comment (cyanglaz) [reply packages/flutter_tools/lib/src/commands/create.dart:136]: I see. Maybe `skip-name-checks`? And add a "help" message to explain such?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/create.dart]: done, though it is still hidden since it is only for internal use

**Review by cyanglaz — APPROVED** [review]
LGTM

**Pull request merged**